lib/parse.js

```diff
@@ -1940,8 +1944,9 @@ function parse($TEXT, options) {
         var first = true, a = [];
         while (!is("punc", closing)) {
             if (first) first = false; else expect(",");
-            if (allow_trailing_comma && is("punc", closing)) break;
-            if (is("punc", ",") && allow_empty) {
+            if ((
+                (allow_trailing_comma && is("punc", closing)) || is("punc", ",")
```

Comment:
Also I would claim this broke few test262 tests, but I'll run the tests some time later today.